sound/soc/sof/intel/hda.c

```diff
@@ -958,6 +958,31 @@ int hda_dsp_remove(struct snd_sof_dev *sdev)
 	return 0;
 }
 
+int hda_dsp_shutdown(struct snd_sof_dev *sdev)
+{
+	const struct sof_dsp_power_state target_state = {
+		.state = SOF_DSP_PM_D0,
+		.substate = SOF_HDA_DSP_PM_D0I0,
+	};
+	int ret = 0;
+
+	/*
+	 * turn DSP back to D0 from D0I3 before shutting down it,
+	 * to avoid booting fail at some platforms from D0i3 status.
```

Comment:
Would be good to split the Intel specific part into a separate patch. I.e. one to add shutdown op, and another to add Intel HDA implementation of it.

The comment is a bit vague, maybe just leave it out and explain the change in commit.